tmail-backend/event-bus-redis/src/main/scala/org/apache/james/events/RedisEventBusClientFactory.scala

```diff
@@ -30,13 +30,9 @@ class RedisEventBusClientFactory @Singleton() @Inject()
 (redisConfiguration: RedisConfiguration, redisClientFactory: RedisClientFactory) {
   val rawRedisClient: AbstractRedisClient = redisConfiguration match {
     case standaloneConfiguration: StandaloneRedisConfiguration => redisClientFactory.createStandaloneClient(standaloneConfiguration)
-    case masterReplicaRedisConfiguration: MasterReplicaRedisConfiguration =>
-      redisClientFactory.createStandaloneClient(new StandaloneRedisConfiguration(masterReplicaRedisConfiguration.redisURI.value.last,
-        masterReplicaRedisConfiguration.useSSL, masterReplicaRedisConfiguration.mayBeSSLConfiguration, masterReplicaRedisConfiguration.ioThreads, masterReplicaRedisConfiguration.workerThreads))
+    case masterReplicaRedisConfiguration: MasterReplicaRedisConfiguration => redisClientFactory.createMasterReplicaClient(masterReplicaRedisConfiguration)
     case clusterRedisConfiguration: ClusterRedisConfiguration => redisClientFactory.createClusterClient(clusterRedisConfiguration)
-    case sentinelRedisConfiguration: SentinelRedisConfiguration =>
-      redisClientFactory.createStandaloneClient(new StandaloneRedisConfiguration(sentinelRedisConfiguration.redisURI,
-        sentinelRedisConfiguration.useSSL, sentinelRedisConfiguration.mayBeSSLConfiguration, sentinelRedisConfiguration.ioThreads, sentinelRedisConfiguration.workerThreads))
+    case sentinelRedisConfiguration: SentinelRedisConfiguration => redisClientFactory.createSentinelClient(sentinelRedisConfiguration)
   }
```

Comment:
IMO it would make more sense to @Inject the AbstractRedisClient and not the RediClientFactory and the conf to know the actual topology:

 - Less code duplication
 - We would only use a single driver event loop and not one per use case.

Reply:
Still requires refactoring on james then, but that goes with your proposition on James side PR. I will take a look at it.